var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class Babelstream(CMakePackage):
+    """Measure memory transfer rates to/from global device memory on GPUs.
+    This benchmark is similar in spirit, and based on, the STREAM benchmark for CPUs."""
+
+    homepage = "https://github.com/UoB-HPC/BabelStream"
+    git = "https://github.com/UoB-HPC/BabelStream.git"
+    url = "https://github.com/UoB-HPC/BabelStream/archive/refs/tags/v4.0.tar.gz"
+
+    maintainers = ["rob64j", "tomdeakin", "tom91136"]
+
+    version("main", branch="main")
+    version("develop", branch="develop")
+    version("4.0", sha256="a9cd39277fb15d977d468435eb9b894f79f468233f0131509aa540ffda4f5953")
+
+    variant(
+        "build_type",
+        default="Release",
+        description="CMake build type",
+        values=("Debug", "Release"),
+    )
+
+    variant(
+        "model",
+        values=[
+            "std-data",
+            "std-indices",
+            "std-ranges",
+            "ocl",
+            "omp",
+            "hip",
+            "cuda",
+            "kokkos",
+            "sycl",
+            "sycl2020",
+            "acc",
+            "raja",
+            "tbb",
+            "thrust",
+        ],
+        default="std-data",
+        description="Specify the model to be built",
+    )
+
+    variant("flags", values=str, default=" ", description="Additional CXX flags to be provided")
```

Comment:
This might need to be reviseted after we merge #29761 Can you add a TODO on top of the variant?